examples/sampling-slice-hyperrectangles-mcmc.ipynb

```diff
@@ -0,0 +1,940 @@
VOjxZwIxTYNfu8FCsW7rrKDkZYRmkQAwgqmRNv5g0RSNfUEznq5CnNz6VUYEmLRMpELzXxR3VmUJ5Xz208GW6a9l1CiMuB5VLKHUKIaRSszKBQTBwqxkrxQWHv3r1s376d2tpaysvL6evrQ9dHWVx2AvnEJz7BHXfcgRCC+vp6/u7v/o7Tp08PfeL48X8JIe4BrpBS/rMQYgmwUEo5eEGviUYIhOYDyxYbkk7GrypKSWEwQ5bjc5SWtlS0YBOacK1X2Nn3yrwinqRFhBDMwR/V6Y+lqC7PFn4axAANPQPcvmBJbtNZ9ISTWVYQl3qtl6WOq9alwSMkPHJNOFk4qD1XDPUKQ2GZed5TpoUp8+t2GaaVUeVMW7wNiCTlHgWvxEpC9AiJ0kuIp2aRu1TvJ06p6aPK7ElnNtMxmZ6yLWDebHQxK0SF6xCXbibTXovItqhoMr/Ya/a5NpYEBprzrg8g2PAusuxqmkWQgmpqAQEyJhNUYjIQr6NDi3CJVUEwpufdN5mjAHrvh6tcCCyQFovCx3Kuxd7vK2J3lEgMLPypFkojx7HrIzimKykpM6MsChzCP/NDxH0eN81hCN+F8I7Ce51ho4P5eieWc0BjbFcmNsh5iVuepBqe+fS2eSTidzYUVhriwkBzdumpoa1wuchjL3BJqIBVWWhpK2kCAyjNi4nao7XzITmnoHvp8DovHOEnc14L4s6XzNQKc8ldXtDF4K6uhfBZSaal/CQEnu+1rWAFSVLpqDUS6A7GMaM6DZHIoPdqvNCGPgSEEPcCLwBPOpsuBV4er0EpFAqFIkNnZyc1NTWUl9v++zU1NSxaZAfBP/jgg6xdu5brrruOjRs3poWFDRs28O1vf5tbbrmFFStWcPDgQe666y6WL1/Od7/7XcAuNHzNNdfwpS99idWrV3P33XcTi8Xy+n/ttddYv349N9xwA5/73OeIROw0APfffz/XXnstq1ev5jvf+U7eeVVVVWkhORqNTujqYRF+DKwH/hfnc8TZNuUQmi0oxD3pxC3sNMqNveF0Brr+mJ6W/Ao7/9iUGV7rkI1mGfQ7K/nFVpePJe26TbqZLwy1B+MFlats8vcXewpy3d68x6WsTP/5MSM2cY8LnzlQuKZOdbKD6mQni8LHONEezBufKfKdqPZrnekthmXHlphS0hr4N2oCb3NGH+AX0ROcEf2DZiPMdmHMbM+d+3ozE7/j7vETp49Y2lrQK+J2zam8XjKz9o5oI4XJO9YJ3hAXOCVsr4WecIK4YZHy3DvDknkKlpc2YVsQhZTMi57lssB+kp54rspUgLmxJjRPdsZcwT+RE09V4rizyvRnWyQdsDKeFprMF1NjKZNYykw/e/mum/l0BW330xlRe4HHvXRTJulw0pq4I+nVG6lMDeS14X0e3fi5mYmh87HZsXrZcyKRNM9eX/QcE4nPylmIiPVj5NRFTxgmYY/S795TvWjs2tC/wRkXwexjfTkaUyE1TKYV8WwXU3B+q0aJzLJW5SOk5JjoTe+NJk1SznORScIzBRUsbPe9m4EQgJTyLDB/0DMUCsWUZnHoMItDh1nXsiXrj53ft/8UU4ZPfvKTtLa2cvXVV7Np0yZ27dqV3vfNb36TgwcPcvz4ceLxONu3b0/vKysrY/fu3Xzta1/jzjvv5Mc//jHHjx9n69at+P22sHXmzBk2btxIfX091dXVPPFEtvtjX18fmzdvZseOHdTW1rJmzRp+9KMf0d/fz0svvcSJEyeor69PK225vPTSS1xzzTV86lOf4he/+MU4zM6IuElK+Q2wg32klAOkE3dPNWyBwu8RStJ1hrCtWS6hRP4Kec6iPFV6JqbLFTgqvDErgyi/SdOiu0Ba5yF1qyIUU7Szs5xJop6EC9l92SvWQuYK1h7XskjxNNQu03U/WkFBNP/C3FTYrwfaaA/G6XAE9rdFG/8eshMinBH9Bc8dCq+dSArotPKtknu0dt7VOpyx2IJ3KGHQFojTFXJj17JdpfpFgkOiG4TAEJmqVIXuWyCeQkornf3Qbq2Q+manLw+Q4M9aM22EKTXjzI2fo9IYoAQNUzrZCz2n94gYO7WWrPFNl9k2OFdhC2WVDCj8rPijmWMG4tmKSAOF7oOkOtlJpZGrOEne1NsIJQzmhk5RoduW0VS6xNQAACAASURBVLnxprzewwVyalYaA1kKTiE0mXmOvfXXUlp+BsKobh9ruQW2PXSFEpR6MlEiIZIsrEjZii704darcmbEilGv9RY8x0vuNVUYIUp9mSyPSUx6yF+My4zZLvidvW9w4h6leSRM1/sKbM0slDSKwLD6H2uGG4OVlFLq7o+iEKKEiR+rQqEYB3JjuNKJMF6ywyg3zVpd+MS/nPqhLO8XqqqqOHz4MG+//TY7d+7k85//PD/4wQ/48pe/zM6dO/nhD39ILBajv7+flStX8pnPfAaAO+6w8xCtWrWKlStXsnChXZho2bJltLa2MmvWLJYsWcLNN98MwD333MNjjz2WZY3at28fJ0+eTB+j6zrr16+nurqaiooKvvrVr/KpT32KT3/60wXH/tnPfpbPfvaz7N69mwceeIAdO3aM2zwNg5STOdBenxXC9lSaggizuIIgkcyQZQVkz1yRjLSmlWtNyOuvaF+D18wZipEICl6B3vKI+cVwI65CCYPyEi2dJAKyY0qSBaxvMWFnJ7tMl5zL3VmgyxJHsM1Nc64Li95IknKfm61t5KJRRqjM3CtTSjqCiYL3pSaW/ZvtWi18Vgp/w1HmTs+sGfSKGAwZxeR2n8qaq2JXckjfx02OspEUJjOTGSuO6yI4LdWP9JUgsdN2F8ItDuxef7fIF9hLhm0HAImGwOK01g/MydpXbkSoTuZbm1yLYzCRolSkaOn5AVRnEot6rYuJIu6KgfjgdZyW+t+g28mz5x7bJIJcU8Da2R/T0UQZFpL+nCQwqTyXzqHtUYbjhheI64Ck1jw1HCMWiZSJ6RlfdbIDsyqTxfDP2vnCJ4qMgj
```

Comment:
Never say something's obvious! (or trivial etc.)
If readers get it you're wasting space, if they don't you're making them feel bad